File: wlxx-framework/wlxx-core/src/main/java/com/xzg/wlxx/core/base/domain/po/BasePo.java
package com.xzg.wlxx.core.base.domain.po;

import com.baomidou.mybatisplus.annotation.*;
import com.baomidou.mybatisplus.extension.activerecord.Model;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author xzgang0499
 * @date 2021-12-27
 * @since jdk1.8
 */

@EqualsAndHashCode(callSuper = true)
@Data
public class BasePo<T extends Model<?>> extends Model<T> {

    @Schema(description = "主键")
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    // TODO 独立日志记录
    @Schema(description = "创建者")
    @TableField(value = "creator", fill = FieldFill.INSERT)
    private Long creator;

    @Schema(description = "创建时间")
    //@JsonFormat(timezone = "GMT+8", pattern = "yyyy-MM-dd HH:mm:ss")
    //@DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME, pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField(value = "create_time", fill = FieldFill.INSERT)
    private Long createTime;

    @Schema(description = "更新人")
    @TableField(value = "updater", fill = FieldFill.UPDATE)
    private Long updater;

    @Schema(description = "更新时间")
    //@JsonFormat(timezone = "GMT+8", pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField(value = "update_time", fill = FieldFill.UPDATE)
    //@DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME, pattern = "yyyy-MM-dd HH:mm:ss")
    private Long updateTime;

    @Schema(description = "是否已删除")
    @TableField(value = "deleted", fill = FieldFill.INSERT)
    @TableLogic
    private Integer deleted;
}

File: wlxx-common/src/main/kotlin/com/xzg/wlxx/common/base/BasePo.kt
package com.xzg.wlxx.common.base

import com.baomidou.mybatisplus.annotation.FieldFill
import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.extension.activerecord.Model
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDateTime


@Schema(name = "PO基础")
open class BasePo<T : BasePo<T>>(
    @TableId("id", type = IdType.AUTO)
    @Schema(name = "ID")
    var id: Long? = null,
    @TableField("create_date", fill = FieldFill.INSERT)
    var createDate: LocalDateTime? = null,
    @TableField("create_by", fill = FieldFill.INSERT)
    var createBy: String? = null,
    @TableField("update_date", fill = FieldFill.UPDATE)
    var updateDate: LocalDateTime? = null,
    @TableField("update_by", fill = FieldFill.UPDATE)
    var updateBy: String? = null
) : Model<T>()